### PR TITLE
Log profile events send timings

### PR DESCRIPTION
**Pull request by davenger**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
On big clusters profile events blocks that are sent back to user might become big and sending them might take noticeable time if user connection is slow.
```
2024.04.16 14:12:12.683778 [ 4592 ] {75fe800a-0dd7-4b29-a360-dc26e9592bee} <Trace> TCPHandler: Sending profile events block with 2815214 rows, 428169479 bytes took 145713 milliseconds
```

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 01b80e754abbde6e379064bdd01987b1990a2b42 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62752/01b80e754abbde6e379064bdd01987b1990a2b42/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Server/TCPHandler.cpp [hunk 1]
```diff
@@ -1244,6 +1245,9 @@ void TCPHandler::sendProfileEvents()
 
         state.profile_events_block_out->write(block);
         out->next();
+
+        LOG_TRACE(log, "Sending profile events block with {} rows, {} bytes took {} milliseconds",
```
Comment (nickitat) [inline src/Server/TCPHandler.cpp:1249]: would't it be too talkative?

Comment (davenger) [reply src/Server/TCPHandler.cpp:1249]: In one of the test runs I see
```
davenger@davenger-xps:~/Downloads$ zstdcat clickhouse-server\ \(7\).log.zst | wc -l
1790422
davenger@davenger-xps:~/Downloads$ zstdcat clickhouse-server\ \(7\).log.zst | grep -a 'Sending profile events' | wc -l
46666
```

Do you think it makes sense to move it under smth like `if (elapsed_milliseconds > 100)` ?

Comment (nickitat) [reply src/Server/TCPHandler.cpp:1249]: yeah, most likely it is useless if the elapsed time is small

File: src/Server/TCPHandler.cpp [hunk 2]
```diff
@@ -1244,6 +1245,9 @@ void TCPHandler::sendProfileEvents()
...
+            block.rows(), block.bytes(), stopwatch.elapsedMilliseconds());
```
Comment (davenger) [inline src/Server/TCPHandler.cpp:1250]:
```suggestion
        auto elapsed_milliseconds = stopwatch.elapsedMilliseconds();
        if (elapsed_milliseconds > 100)
            LOG_DEBUG(log, "Sending profile events block with {} rows, {} bytes took {} milliseconds",
                block.rows(), block.bytes(), elapsed_milliseconds);
```

Comment (nickitat) [reply src/Server/TCPHandler.cpp:1250]: maybe as a warning

**Review by nickitat — APPROVED** [review]

**Pull request merged**